### ISSUE TITLE
TSC Meeting (13 Feb 2024)

### ISSUE THREAD

**Issue by nzlosh** [StackStorm/discussions#131]
February 2024 @StackStorm/tsc 1 hour meeting:

  - Tuesday, 13 February 2024, 09:30 AM US Pacific / 06:30 PM EU CET
  - See https://github.com/StackStorm/community/issues/33 for the schedule and how to join
  - TLDR; Jitsi link: https://meet.jit.si/StackStormTSC

# Meeting Agenda

## StackStorm v3.9.0

https://github.com/orgs/StackStorm/projects/31

- *Target:* ????
- *Release Manager:* TSC volunteer Needed!
- *Release Assistant:* TSC volunteer Needed!

## Add Support for MongoDB 5, 6, 7

https://github.com/StackStorm/st2/pull/6079

Progress update?
Assistance required?

## Eventlet Retirement

https://github.com/StackStorm/st2/issues/6128

Eventlet team has announced its intention to retire eventlet.  They recommend migrating to Python's standard asyncio solution and have stated they'll provide a migration path for their users which is under active development.

Discussion on slack mentioned migrating to gevent or re-write using asyncio.

- 701 references to `eventlet`
- 193 files reference `eventlet`
  - 41 files reference `import eventlet`

Release target: ?
Developer Lead: ?

 ## StackStorm Exchange - Pack compatibility with Python3.9/3.10/3.11

With the addition of py3.9, 3.10 and 3.11 to StackStorm, the exchange packs will need to be tested for compatibility.  This will include any python code in the pack itself as well as module dependencies.

Release target: ?
Developer Lead: ?

**Comment by nzlosh** [comment]
TSC Minutes (13 Feb 2024)

## Attendees

- @nzlosh
- @ravi
- @mamercad
- @khaddict
- @jk464
- @fuhrmannb
- @cognifloyd
- @winem 
- Dale Smith
- and anothers - sorry, I wasnt able to get everyones names.

## Agenda items

### StackStorm v3.9.0

@amanda, @nzlosh, @khaddict and others have been making progess on the package builds for Ubuntu and Rocky as well as removing older unsupported distributions.

The CircleCI is failing for Rocky9 due to a newer version of Docker/Docker Compose being used.  @nzlosh is working on fixing this.

@mamercad has offered to help with updating the anisble deployment playbooks.  We still require someone to help with the puppet and oneliner install script.

### Add Support for MongoDB 5, 6, 7

The `pymongo` depends on `mongoengine` so they have to be updated together, which involves updating function calls through the code base beyond those that were identified for `pymongo`.

@winem has offered to help with the mongo PR work.

### Eventlet Retirement

`ravi` mentionned that @guzzijones has been looking into alternative concurrency solutions and we should reach out to get more information on what's being done.

gevent appears to be the easiest way forward but we will still watch what eventlet offers as a migration plan.

@cognifloyd mentioned the concurrency stub code that can be used.  It could be possible to re-write async code by component and upgrade StackStorm part by part by creating async and sync versions in parallel.

### StackStorm Exchange - Pack compatibility with Python3.9/3.10/3.11

Packs need to be tested for compatibility but it is unknown who actively maintains to carry out this work.  To help determine if a pack has an active maintainer we will create an issue for each pack requests a response from the maintainer.

`ravi` requested a review for a PR outstanding for the SQL pack  which highlights the lack of pack maintainers.

## Agenda additions

### Pants

@cognifloyd continue to make progress on pants integration.  A draft PR is open to create a pants virtualenv with tests passing.

The PR must be have all tests passing with nose before migrating them to pytest.   Almost all tests are passing now.


### Pack testing bug

@cognifloyd found a bug in the CI that has broken pack testing for at least a year. :(  When the tests were set up in Github actions, an environment variable was used for a logic test which cause pack tests to be skipped.

### Website update

`Dale Smith` has a local setup running for the new web content stack

 - Astro https://astro.build/
 - Netlify https://www.netlify.com/ for content creation
 - Figma https://www.figcomponents.com/

There a question around how to share components.  Dale will set up an initial configuration and the rest of the community can provide feedback for adjustments if necessary.

Since the scope of this work has shifted from the initial issue https://github.com/StackStorm/community/issues/123, a new new issue wil be created to track progression forward.

### Kubernetes/Docker PRs

@jk464 Has multiple Kubernetes and Docker PRs open to address security fixes.  These PR need to be reviewed by a maintainer with sufficient experience with Kubernetes/Docker.  It may be necessary to make some changes configurable as they may be specific to the 


## Next steps

1. @mamercad has opened https://github.com/StackStorm/st2/issues/6135 to fix the *Pack testing bug*
2. Maintainer needed to reivew Kubernetes/Docker PRs.
  - https://github.com/StackStorm/stackstorm-k8s/pull/401
  - https://github.com/StackStorm/stackstorm-k8s/pull/400
  - https://github.com/StackStorm/stackstorm-k8s/pull/399
  - https://github.com/StackStorm/stackstorm-k8s/pull/398
  - https://github.com/StackStorm/stackstorm-k8s/pull/397
  - https://github.com/StackStorm/stackstorm-k8s/pull/396
  - https://github.com/StackStorm/st2-dockerfiles/pull/66
3. Maintainer needed to review https://github.com/StackStorm-Exchange/stackstorm-sql/pull/30
